pyhf/pdf.py

```diff
@@ -15,6 +16,11 @@ def from_spec(cls,spec,poiname = 'mu'):
         for channel in spec['channels']:
             for sample in channel['samples']:
                 for modifier_def in sample['modifiers']:
+                    modifier_def = copy.deepcopy(modifier_def)
```

Comment:
I copy here in order to not changes the original object being passed in by the user